pkg/metrics/metric_sample.go

```diff
@@ -149,3 +150,19 @@ func (m *MetricSample) Copy() *MetricSample {
 	copy(dst.Tags, m.Tags)
 	return dst
 }
+
+// taggerCardinality converts tagger cardinality string to collectors.TagCardinality
+// It defaults to DogstatsdCardinality if the string is empty or unknown
+func taggerCardinality(cardinality string) collectors.TagCardinality {
```

Comment:
I know this has been discussed with @hush-hush, but I'm not sure doing the `string` to `int` (`collectors.TagCardinality`) conversion here instead of earlier in the pipeline will actually improve performance: doing the string conversion here means that the `MetricSample` has to keep a string instead of just an `int` value.

Since I believe this string is interned in the string interner, it doesn't put much additional pressure on the GC (I think, but not sure), so it may not have any noticeable impact, but worth discussing. Unfortunately I can't think of a good micro-benchmark to measure the impact.

In any case this shouldn't block the PR, but happy to hear your thoughts @hush-hush

Reply:
The string should be in the interner. And doing it later in the pipeline allows us to use the channel between the aggregator and dogstatsd to absorb spike instead of holding a `packet` from the packet pool longer.

Reply:
ok, thanks for chiming in and confirming this string is interned. I think it's hard to predict the impact without an e2e benchmark though, but not a blocker now

Reply:
thank you both! @olivielpeau unless you have other suggestions, could you add your +1 so we can merge the PR?